src/slurm_plugin/clustermgtd.py

```diff
@@ -628,8 +632,22 @@ def _handle_unhealthy_dynamic_nodes(self, unhealthy_dynamic_nodes):
             self._instance_manager.delete_instances(
                 instances_to_terminate, terminate_batch_size=self._config.terminate_max_batch_size
             )
-        log.info("Setting unhealthy dynamic nodes to down and power_down.")
-        set_nodes_power_down([node.name for node in unhealthy_dynamic_nodes], reason="Scheduler health check failed")
+        if not self._is_smart_instance_capacity_fail_over_enabled():
+            log.info(
+                "Setting the following unhealthy dynamic nodes to down and power_down: %s",
+                print_with_count(unhealthy_dynamic_nodes),
+            )
+            set_nodes_power_down(
+                [node.name for node in unhealthy_dynamic_nodes], reason="Scheduler health check failed"
+            )
+        else:
+            unhealthy_not_ice_nodes = [node.name for node in unhealthy_dynamic_nodes if not node.is_ice()]
+            if unhealthy_not_ice_nodes:
+                log.info(
+                    "Setting the following unhealthy dynamic nodes to down and power_down: %s",
+                    print_with_count(unhealthy_not_ice_nodes),
+                )
+                set_nodes_power_down(unhealthy_not_ice_nodes, reason="Scheduler health check failed")
```

Comment:
you could do:
```
if not self._is_smart_instance_capacity_fail_over_enabled():
  nodes_to_power_down = [node.name for node in unhealthy_dynamic_nodes]
else:
  nodes_to_power_down = [node.name for node in unhealthy_dynamic_nodes if not node.is_ice()]

log.info(
                "Setting the following unhealthy dynamic nodes to down and power_down: %s",
                print_with_count(unhealthy_dynamic_nodes),
            )
set_nodes_power_down(
                nodes_to_power_down, reason="Scheduler health check failed"
            )
```

Reply:
Changed to
```
        nodes_to_power_down = (
            [node for node in unhealthy_dynamic_nodes if not node.is_ice()]
            if self.disable_nodes_on_insufficient_capacity
            else unhealthy_dynamic_nodes
        )

        if nodes_to_power_down:
            log.info(
                "Setting the following unhealthy dynamic nodes to down and power_down: %s",
                print_with_count(nodes_to_power_down),
            )
            set_nodes_power_down([node.name for node in nodes_to_power_down], reason="Scheduler health check failed")
```
With ICE nodes, it is possible that nodes_to_power_down is empty.